ipynb/examples/utils/gem5_example.ipynb

```diff
@@ -0,0 +1,570 @@
+      "2017-05-23 22:19:20,035 INFO    : TestEnv      :    sched_load_avg_task\n",
+      "2017-05-23 22:19:20,035 INFO    : TestEnv      :    sched_load_waking_task\n",
+      "2017-05-23 22:19:20,035 INFO    : TestEnv      :    cpu_capacity\n",
+      "2017-05-23 22:19:20,036 INFO    : TestEnv      :    cpu_frequency\n",
+      "2017-05-23 22:19:20,036 INFO    : TestEnv      :    cpu_idle\n",
+      "2017-05-23 22:19:20,036 INFO    : TestEnv      :    sched_energy_diff\n",
+      "2017-05-23 22:19:20,037 INFO    : TestEnv      : Set results folder to:\n",
+      "2017-05-23 22:19:20,037 INFO    : TestEnv      :    /home/vagrant/results/gem5_res\n",
+      "2017-05-23 22:19:20,037 INFO    : TestEnv      : Experiment results available also in:\n",
+      "2017-05-23 22:19:20,038 INFO    : TestEnv      :    /home/vagrant/results_latest\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Create the hardware target. Patience is required :\n",
+    "# ~40 minutes to resume from a checkpoint (detailed)\n",
+    "# ~5 minutes to resume from a checkpoint (atomic)\n",
+    "# ~3 hours to start from scratch (detailed)\n",
+    "# ~15 minutes to start from scratch (atomic)\n",
+    "te = TestEnv(conf)\n",
+    "target = te.target"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "deletable": true,
+    "editable": true
+   },
+   "source": [
+    "# Run workloads on gem5"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This is an example of running a workload and extracting stats from the simulation using m5 commands. For more information about m5 commands, see http://gem5.org/M5ops"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "collapsed": false,
+    "deletable": true,
+    "editable": true,
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2017-05-23 22:19:49,308 DEBUG   : ssh          : Pushing /home/vagrant/tools/arm64/sysbench to device.\n",
+      "2017-05-23 22:19:49,309 DEBUG   : ssh          : gem5interactdir: /tmp/wa_4\n",
+      "2017-05-23 22:19:49,310 DEBUG   : ssh          : dest: /home/root/devlib-target/bin/sysbench\n",
+      "2017-05-23 22:19:49,310 DEBUG   : ssh          : filename: sysbench\n",
+      "2017-05-23 22:20:02,329 DEBUG   : ssh          : Push complete.\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "''"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Initialise command: [binary/script, arguments]\n",
+    "command = 'sysbench'\n",
+    "args = '--test=cpu --max-time=1 run'\n",
+    "\n",
+    "# Install binary if needed\n",
+    "path = target.install_if_needed(\"/home/vagrant/tools/arm64/\" + command)\n",
+    "\n",
+    "# Reset and dump statistics\n",
+    "target.execute('m5 resetstats')\n",
```

Comment:
Is there any gem5 support in devlib?

If there is one, these commands (i.e. resetstats/dumpstats) should be better provided via that API.
In case they are not yet available, they can be added there.

In general we don't really want a user has to remember device specific commands...